s3-handler/main.go

```diff
@@ -71,7 +71,7 @@ func Handler(request events.S3Event) (Response, error) {
 				}).Info("parser checkpoint")
 			}
 
-			if sampleRate != 1 && rand.Intn(int(sampleRate)) != 0 {
+			if sampleRate != 0 && sampleRate != 1 && rand.Intn(int(sampleRate)) != 0 {
```

Comment:
I think something is wrong in [common.InitHoneycombFromEnvVars()](https://github.com/honeycombio/agentless-integrations-for-aws/blob/main/common/common.go#L38-L46) if `sampleRate` is getting set to 0?

Reply:
Looks like `SAMPLE_RATE=0` would get through that function just fine. It does probably mean that this fix would be better in `InitHoneycombFromEnvVars` though — so all callers of that would be guaranteed a non-zero uint.

Reply:
thank you!! 👁️ to update this now <3

Reply:
Can't this still be set to zero: we're now going to log a message if `unit(i)` is zero but right below we set sample rate to the value regardless. Do we need an `else`?

Reply:
oooh omg yeah i am not a bright bulb thank you!